File: ruby/darts/darts.rb
# frozen_string_literal: true

class Darts
  def initialize(x, y)
    @x = x
    @y = y
    @r = Math.sqrt(@x**2 + @y**2)
  end

  def score
    if @r <= 1
      10
    elsif @r > 1 && @r <= 5
      5
    elsif @r > 5 && @r <= 10
      1
    else
      0
    end
  end
end

File: r/darts/darts.R
score <- function(x, y) {
  coord = sqrt(x^2 + y^2)
  if (coord > 10) {
     0
  } else if (coord <= 10 && coord > 5) {
    1
  } else if (coord <= 5 && coord > 1) {
    5
  } else {
    10
  }
}
